[BUG] Allow secret to be set even though auth is disabled

```
Traceback (most recent call last):
  File "<frozen runpy>", line 189, in _run_module_as_main
  File "<frozen runpy>", line 112, in _get_module_details
  File "/Users/mikeldking/work/phoenix/src/phoenix/__init__.py", line 1, in <module>
    from .inferences.fixtures import ExampleInferences, load_example
  File "/Users/mikeldking/work/phoenix/src/phoenix/inferences/fixtures.py", line 12, in <module>
    from phoenix.config import INFERENCES_DIR
  File "/Users/mikeldking/work/phoenix/src/phoenix/config.py", line 184, in <module>
    ENABLE_AUTH, PHOENIX_SECRET = get_auth_settings()
                                  ^^^^^^^^^^^^^^^^^^^
  File "/Users/mikeldking/work/phoenix/src/phoenix/config.py", line 154, in get_auth_settings
    assert not phoenix_secret, (
           ^^^^^^^^^^^^^^^^^^
AssertionError: `DANGEROUSLY_SET_PHOENIX_SECRET` must not be set unless auth is enabled with `DANGEROUSLY_SET_PHOENIX_ENABLE_AUTH`
```

We may still want to use the secret for symetric signing of data.

--- a/src/phoenix/config.py
+++ b/src/phoenix/config.py
@@ -150,14 +150,9 @@ def get_auth_settings() -> Tuple[bool, Optional[str]]:
     """
     enable_auth = get_env_enable_auth()
     phoenix_secret = get_env_phoenix_secret()
-    if enable_auth:
-        assert phoenix_secret, (
+    if enable_auth and not phoenix_secret:
+        raise ValueError(
             f"`{ENV_PHOENIX_SECRET}` must be set when "
-            f"auth is enabled with `{ENV_PHOENIX_ENABLE_AUTH}`"
-        )
-    else:
-        assert not phoenix_secret, (
-            f"`{ENV_PHOENIX_SECRET}` must not be set unless "
             f"auth is enabled with `{ENV_PHOENIX_ENABLE_AUTH}`"
         )
     return enable_auth, phoenix_secret

